src/cgame/cg_view.cpp

```diff
@@ -1414,10 +1414,27 @@ static void CG_CalcColorGradingForPoint( vec3_t loc )
 		}
 	}
 
-	cg.refdef.gradingWeights[0] = 0.0f;
-	cg.refdef.gradingWeights[1] = haveGlobal ? ( 1.0f - totalWeight ) : ( selectedWeight[0] / totalWeight );
-	cg.refdef.gradingWeights[2] = totalWeight == 0.0f ? 0.0f : selectedWeight[1] / totalWeight;
-	cg.refdef.gradingWeights[3] = totalWeight == 0.0f ? 0.0f : selectedWeight[2] / totalWeight;
+	if ( totalWeight )
+	{
+		cg.refdef.gradingWeights[0] = 0.0f;
+		cg.refdef.gradingWeights[1] = selectedWeight[0];
+		cg.refdef.gradingWeights[2] = selectedWeight[1];
+		cg.refdef.gradingWeights[3] = selectedWeight[2];
+
+		cg.refdef.gradingWeights[0] /= totalWeight;
+		cg.refdef.gradingWeights[1] /= totalWeight;
+		cg.refdef.gradingWeights[2] /= totalWeight;
+		cg.refdef.gradingWeights[3] /= totalWeight;
+	}
+	else
+	{
+		Vector4Set( cg.refdef.gradingWeights, 0, 0, 0, 0 );
+	}
+
+	if ( haveGlobal )
+	{
+		cg.refdef.gradingWeights[1] = 1.0f - totalWeight ;
```

Comment:
Why is this needed now?

Reply:
This code will be obsoleted by: https://github.com/Unvanquished/Unvanquished/pull/3213

The code was doing that before, but without a division by zero for free. I don't know what this code does.